### PR TITLE
Export godep patch files to artifacts

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
If a godep patch file is created, and a `${WORKSPACE}/_artifacts` directory exists, copy the patch file out to it.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

cc: @fejta

### CODE REVIEW

**Comment by eparis** [comment]
/approve
but I'll let @fejta do the lgtm

**Comment by cblecker** [comment]
@eparis is it possible to also get an `okay to test`? thank you!

**Comment by eparis** [comment]
@k8s-bot ok to test

**Comment by cblecker** [comment]
@k8s-bot non-cri node e2e test this
@k8s-bot gci gce e2e test this
@k8s-bot non-cri e2e test this
@k8s-bot cvm gce e2e test this

**Comment by cblecker** [comment]
@fejta Tests passing -- ready for review! 🎉

**Comment by fejta** [comment]
Can you create another PR that a) has this commit and b) includes some godep changes?

It does not look like we hit your changes (click build log and/or artifacts):
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/43020/pull-kubernetes-verify/21898/

**Comment by fejta** [comment]
Also from looking at https://github.com/kubernetes/test-infra/blob/master/scenarios/kubernetes_verify.py#L86 it looks like we may either need to change this to be $WORKSPACE/artifacts or else change the scenario to mount _artifacts outside the container to _artifacts inside...

**Comment by cblecker** [comment]
Ahh okay, I see it. It looks like it mounts `$WORKSPACE/_artifacts` to `/workspace/artifacts` in the container, if I'm reading that correctly?

**Comment by fejta** [comment]
Probably best to just mount it in the same location...

File: hack/verify-godeps.sh [hunk 1]
```diff
@@ -128,7 +128,12 @@ pushd "${KUBE_ROOT}" 2>&1 > /dev/null
     echo "If you're seeing this locally, run the below command to fix your Godeps.json:"
     echo "patch -p0 < godepdiff.patch"
     echo "(The above output can be saved as godepdiff.patch if you're not running this locally)"
+    echo "(The patch file should also be exported as a build artifact if run through CI)"
     KEEP_TMP=true
+    if [[ -x godepdiff.patch && -d "${WORKSPACE}/_artifacts" ]]; then
```
Comment (ixdy) [inline hack/verify-godeps.sh]: maybe use `${ARTIFACTS_DIR}` instead.

Comment (cblecker) [inline hack/verify-godeps.sh]: `${ARTIFACTS_DIR}` seems to be exported in https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/test-dockerized.sh, but not in https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/verify-dockerized.sh (which this specific job seems to use https://github.com/kubernetes/test-infra/blob/1a6f132432670e55a1be25bca50b73735f76420e/jobs/config.json#L353)

Comment (fejta) [inline hack/verify-godeps.sh]: I don't think that is set in https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/verify-dockerized.sh?

**Comment by ixdy** [comment]
oh right. I got confused because the `kubernetes_verify.py` defaults to calling `test-dockerized.sh`.

**Comment by ixdy** [comment]
(how is naming SO HARD?)

**Comment by cblecker** [comment]
@ixdy @fejta 
Does this work? I'm now exporting `${ARTIFACTS_DIR}`, and then using it in the script.

File: hack/verify-godeps.sh [hunk 2]
```diff
@@ -128,7 +128,12 @@ pushd "${KUBE_ROOT}" 2>&1 > /dev/null
     echo "If you're seeing this locally, run the below command to fix your Godeps.json:"
     echo "patch -p0 < godepdiff.patch"
     echo "(The above output can be saved as godepdiff.patch if you're not running this locally)"
+    echo "(The patch file should also be exported as a build artifact if run through CI)"
     KEEP_TMP=true
+    if [[ -x godepdiff.patch && -d "${ARTIFACTS_DIR:-}" ]]; then
```
Comment (ixdy) [inline hack/verify-godeps.sh]: are the patch files really executable?

Comment (cblecker) [inline hack/verify-godeps.sh]: *facepalm* they are not. fixing!

**Comment by fejta** [comment]
If you can get it to work in https://github.com/kubernetes/kubernetes/pull/43038 sure!

**Comment by cblecker** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot non-cri e2e test this
@k8s-bot gce etcd3 e2e test this

**Comment by cblecker** [comment]
@fejta Looks like it's working as intended here: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/43038/pull-kubernetes-verify/21935/

**Comment by cblecker** [comment]
kops failures related to https://github.com/kubernetes/kubernetes/issues/43017#issuecomment-286303401

**Comment by cblecker** [comment]
@k8s-bot kops aws e2e test this

**Comment by cblecker** [comment]
@k8s-bot kops aws e2e test this

**Pull request merged**